rust/arrow/src/buffer.rs

```diff
@@ -1003,6 +1156,28 @@ impl PartialEq for MutableBuffer {
 unsafe impl Sync for MutableBuffer {}
 unsafe impl Send for MutableBuffer {}
 
+struct SetLenOnDrop<'a> {
```

Comment:
this construct is fascinating, but I will take your word that it was necessary for speedup